src/components/bookingFirstForm/BookingFirstForm.tsx

```diff
@@ -0,0 +1,140 @@
+import { ChangeEvent, FormEvent, ReactNode, useState } from "react";
+import PhoneInput from 'react-phone-number-input'
+import { FormError } from "../../utils/types";
+import CustomInputWithErrorMessage from "../customInputWithErrorMessage/CustomInputWithErrorMessage";
+import 'react-phone-number-input/style.css'
+
+type BookingFirstFormProps = {
+  setStep(step: (prev: number) => number): void;
+  actions: ReactNode;
+}
+
+const BookingFirstForm = ({ setStep, actions }: BookingFirstFormProps) => {
+  const [formData, setFormData] = useState({
+    name: '',
+    lastName: '',
+    email: '',
+    phoneNumber: ''
+  });
+  const [formErrors, setFormErrors] = useState<Array<FormError> | null>(null);
+
+  const validateEmail = (email: string) => {
+    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(email);
+  };
+
+  const validField = (formNameInput: any, id: string, name: string, error: string): FormError | undefined => {
+    if (!formNameInput.trim()) {
+      return {
+        id,
+        name,
+        error
+      }
+    }
+    if (name === "email" && !validateEmail(formNameInput)) {
+      return {
+        id,
+        name,
+        error: "El correo electrónico no es válido"
```

Comment:
are we going to support multiple languages? if so, returning the error in a specific language is not a good idea.